src/components/themed/FlipInput2.tsx

```diff
@@ -193,86 +194,151 @@ export const FlipInput2 = React.forwardRef<FlipInputRef, Props>((props: Props, r
+    paddingHorizontal: theme.rem(1),
+    paddingVertical: theme.rem(0.5),
+    position: 'absolute',
+    top: 0,
+    left: 0,
+    right: 0,
+    bottom: 0
+  },
+  useAnimatedStyle(() => {
+    const degrees = interpolate(animatedValue.value, [0, 0.5, 1], [90, 90, 0])
+    return {
+      transform: [{ rotateX: `${degrees}deg` }]
+    }
+  })
+])
+
+const PlaceholderText = styled(Text)(theme => () => [
+  {
+    color: theme.textInputPlaceholderColor,
+    fontFamily: theme.fontFaceDefault,
+    fontSize: theme.rem(0.8),
+    includeFontPadding: false
+  }
+])
+
+const AmountAnimatedNumericInput = styledWithRef(AnimatedNumericInput)<{ focusAnimation: SharedValue<number> }>(theme => ({ focusAnimation }) => {
+  const isIos = Platform.OS === 'ios'
+  const interpolateTextColor = useAnimatedColorInterpolateFn(theme.textInputTextColor, theme.textInputTextColorFocused)
+  return [
+    {
       paddingRight: isIos ? 0 : theme.rem(0.25),
-      color: theme.primaryText,
       includeFontPadding: false,
       fontFamily: theme.fontFaceMedium,
       fontSize: isIos ? theme.rem(1.5) : theme.rem(1.45)
     },
-    bottomAmountTappable: {
-      paddingRight: isIos ? 0 : theme.rem(0.25),
-      color: theme.iconTappable,
+    useAnimatedStyle(() => ({
+      color: interpolateTextColor(focusAnimation)
+    }))
+  ]
+})
+const CurrencySymbolAnimatedText = styled(Animated.Text)<{ focusAnimation: SharedValue<number> }>(theme => ({ focusAnimation }) => {
+  const isIos = Platform.OS === 'ios'
+  const interpolateTextColor = useAnimatedColorInterpolateFn(theme.textInputTextColor, theme.textInputTextColorFocused)
+  return [
+    {
+      fontFamily: theme.fontFaceDefault,
+      fontSize: theme.rem(1),
       includeFontPadding: false,
-      fontFamily: theme.fontFaceMedium,
-      fontSize: isIos ? theme.rem(1.5) : theme.rem(1.45)
-    },
-    bottomCurrency: {
       paddingTop: isIos ? theme.rem(0.125) : theme.rem(1),
       marginLeft: isIos ? 0 : -theme.rem(0.25)
     },
-    bottomCurrencyTappable: {
-      paddingTop: isIos ? theme.rem(0.125) : theme.rem(1),
-      marginLeft: isIos ? 0 : -theme.rem(0.25),
-      color: theme.iconTappable
-    }
+    useAnimatedStyle(() => ({
+      color: interpolateTextColor(focusAnimation)
+    }))
+  ]
+})
+
+const BottomContainerView = styled(View)(theme => ({
+  flexDirection: 'row',
+  marginRight: theme.rem(1.5),
+  minHeight: theme.rem(2)
+}))
+
+const ValueContainerView = styled(View)(theme => {
+  const isIos = Platform.OS === 'ios'
+  return {
+    flexDirection: 'row',
+    marginRight: theme.rem(0.5),
+    marginLeft: isIos ? 0 : -6,
+    marginTop: isIos ? 0 : -theme.rem(0.75),
+    marginBottom: isIos ? 0 : -theme.rem(1)
   }
 })
+
+function useAnimatedColorInterpolateFn(fromColor: string, toColor: string) {
+  const interpolateFn = useMemo(() => {
+    return (focusValue: SharedValue<number>) => {
+      'worklet'
+      return interpolateColor(focusValue.value, [0, 1], [fromColor, toColor])
```

Comment:
Please pass the "RGB" color space parameter: https://docs.swmansion.com/react-native-reanimated/docs/utilities/interpolateColor/#arguments

The default is "HSV" interpolation, which looks awful in almost every circumstance. This is probably why you had to change the transparent outline color a few commits ago - there should be no real difference between `#00000000` and `#00f1a200`, but with "HSV" interpolation, getting it wrong produces flashes or random colors.

Reply:
Looks like switching from HSV to RGB didn't have the effect I'm going for. Instead, what's required to get closer to the same effect would be to use HSV and the `useCorrectedHSVInterpolation` option. I think being explicit in the theme is the right move because it gives all the control up to the theme and is less verbose.